File: examples/Pendulum/PendulumInput.m
classdef PendulumInput < LCMCoordinateFrame & Singleton
  
  methods
    function obj=PendulumInput()
      obj = obj@LCMCoordinateFrame('PendulumInput','drake.examples.Pendulum.lcmt_pendulum_u','u',{'torque'});
      obj = obj@Singleton();
    end
  end
end

File: examples/pendulum/pendulum_input.cc
#include "drake/examples/pendulum/pendulum_input.h"

namespace drake {
namespace examples {
namespace pendulum {

const int PendulumInputIndices::kNumCoordinates;
const int PendulumInputIndices::kTau;

const std::vector<std::string>& PendulumInputIndices::GetCoordinateNames() {
  static const drake::never_destroyed<std::vector<std::string>> coordinates(
      std::vector<std::string>{
          "tau",
      });
  return coordinates.access();
}

}  // namespace pendulum
}  // namespace examples
}  // namespace drake
